src/Modules/Render.js

```diff
@@ -0,0 +1,157 @@
+    item.addEventListener('click', (e) => {
+      const element = e.target;
+      element.src = './images/red-hearth.svg';
+      const like = new Likes();
+      like.setLike(Number(element.getAttribute('data-liked'))).then(() => {
+        const child = element.nextElementSibling.firstChild;
+        const count = Number(child.innerHTML);
+        child.innerHTML = count + 1;
+      });
+    });
+  });
+};
+
+const closeModal = () => {
+  const popup = document.querySelector('.show-popup');
+  const close = document.querySelector('.popup-close');
+  close.addEventListener('click', () => {
+    popup.style.display = 'none';
+  });
+};
+
+/**
+ * This function shows the popup when
+ * click the comment button
+ */
+
+const hitComment = () => {
+  const openPopup = document.querySelectorAll('.hit-comment'); /* Comment button */
+  const popup = document.querySelector('.show-popup');
+
+  openPopup.forEach((item) => {
+    item.addEventListener('click', async (e) => {
+      popup.style.display = 'block';
+      const element = e.target;
+      const id = element.getAttribute('data-display');
+      const product = allProducts[id - 1];
+      const comment = new Comments();
+      const data = await comment.getComments(product.id);
+      const template = showComments(data);
+      popup.innerHTML = showPopup(product, template, data.length);
+      closeModal();
+      const submit = document.querySelector('.submit');
+      const name = document.querySelector('.name');
+      const message = document.querySelector('.message');
+      submit.addEventListener('click', async (e) => {
+        e.preventDefault();
+        const comment = new Comments();
+        comment.setComment(id, name.value, message.value).then(async () => {
+          const data = await comment.getComments(product.id);
+          const template = showComments(data);
+          popup.innerHTML = showPopup(product, template, data.length);
+          closeModal();
+        });
+      });
+    });
+  });
+};
+
+/**
+ * This function accept the following type of parameter
+ * @param {number} index and display only a certain range
+ * of data at the beginning and others will bw displayed with
+ * different value of index for pagination
+ */
+const renderTemplate = async (index = 0) => {
+  const start = index * 6;
+  const container = document.querySelector('#app');
+  let end = index * 6 + 6;
+  let template = '<div class="product-container pt-500 pb-500 m-300">';
+  if (end > allProducts.length) {
+    end = allProducts.length;
+  }
+
+  for (let count = start; count < end; count += 1) {
+    const likeCount = countLikes(allLikedItems, allProducts[count].id);
+    /** WE have the function to get the number of comments for specific card at popup section
+     * Here we were also added that but that was causing the application to be slow
+     * That is way we have added Random number of comments to have better UI insted
+     * of just displaying zeros
+     */
```

Comment:
- [ optional ] Great job. It's not best practice to have comments everywhere in your script. I think your functions are already self-explanatory.